### ISSUE TITLE
Default ungrab combination LS+LC+LA+Escape fails with Windows

### ISSUE THREAD

**Issue by vinriviere** [aranym/aranym#66]
ARAnyM's default key combination to ungrab the mouse is LeftShift + LeftControl + LeftAlt + Escape. There is a reminder in the title bar.

But this combination does not really work on Windows (tested with Windows 10 and sdl1 version). This is because it interferes with Windows hotkeys:

- Ctrl+Escape opens the Start menu (this one doesn't seem to interfere)
- Alt+Escape goes backwards in the history of focused windows
- Shift+Alt+Escape goes forward in the history of focused windows

And finally LeftShift + LeftControl + LeftAlt + Escape seems to behave like just Shift+Alt+Escape.

All these combinations still makes the ARAnyM window losing the focus. As a side effect, it also loses the grabbed mouse status. So we can't say it's totally broken. However, in this situation, the ARAnyM window menu does not display "no input grab" as it should. This is the proof that ARAnyM wrongly considers that the mouse is still grabbed.

This is a minor issue, as the user can still ungrab with the middle mouse button (not easily available on laptops) or Pause key (sometimes missing on laptops).

**Comment by th-otto** [comment]
> This is the proof that ARAnyM wrongly considers that the mouse is still grabbed.

That's possible. Would have to trace the events in windows, but i guess that SDL does not get any event when using such system-hotkeys, or does not propagate it to the application.

Should not be too difficult to change the default for this shortcut on windows, but of course that does not help anymore once you have saved the config file. And we have to do something about the pre-generated config file for the snapshots of FreeMint. Any suggestion what would be a better shortcut?

**Comment by vinriviere** [comment]
This shortcut is definitely inappropriate for Windows.

For already generated config files, I would say don't change anything, to avoid more confusion.
But when there is no config file at all, ARAnyM should propose a decent shortcut. The current one seems to be only problematic for Windows. And I don't really like the idea of different default shortcuts on different systems.

At one era (before middle mouse button ungrab) I used the key above Tab to ungrab. It is labeled ` on US keyboards, or ² on French keyboard. This one is generally a good choice for Atari emulators, as Atari machines don't have that key (more precisely: Escape is at that place).

I have no strong opinion, except that the default is bad for Windows.

**Comment by vinriviere** [comment]
Same happens with sdl2 version. It seems that, on Windows, SDL can't prevent the system hotkeys to do their default action.

**Comment by th-otto** [comment]
> For already generated config files, I would say don't change anything

I mean the one that is placed pre-configured in the snapshot archives (the one that you used your video). If the default turns out to be bad, it could be changed there, too.

> I don't really like the idea of different default shortcuts on different systems.

It's not perfect, but the Mac port already uses different shortcuts, for similar reasons (the defaults from unix don't work there, or are already in use by the system).

**Comment by th-otto** [comment]
> It seems that, on Windows, SDL can't prevent the system hotkeys to do their default action.

Yes, if we don't get any event, then there is not much we can do about that.

**Comment by vinriviere** [comment]
> I mean the one that is placed pre-configured in the snapshot archives (the one that you used your video). If the default turns out to be bad, it could be changed there, too.

Yes, in any case that will have to be changed, too. I would prefer the same configuration as the ARAnyM defaults, anyway. Maybe the solution for FreeMiNT is just to remove the HOTKEYS section from its config file, so it always gets the ARAnyM defaults on startup.

> It's not perfect, but the Mac port already uses different shortcuts, for similar reasons (the defaults from unix don't work there, or are already in use by the system).

Oh. So Windows-specific default shortcuts would not be worse.
